pkg/storage/sql/postgres.go

```diff
@@ -0,0 +1,63 @@
+package sql
+
+import (
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/postgres"
+	"github.com/trustwallet/blockatlas/pkg/errors"
+	"github.com/trustwallet/blockatlas/pkg/storage"
+)
+
+type PgSql struct {
+	client *gorm.DB
+}
+
+func (db *PgSql) Init(host string) error {
+	client, err := gorm.Open("postgres", host)
+	if err != nil {
+		return errors.E(err, "postgress connection failed")
+	}
+	db.client = client
+	return nil
+}
+
+func (db *PgSql) Get(value interface{}) error {
+	err := db.client.Last(value).Error
+	if err != nil {
+		return errors.E(err, storage.ErrNotFound)
+	}
+	return nil
+}
+
+func (db *PgSql) Update(value interface{}) error {
+	if db.client.NewRecord(value) {
+		return storage.ErrNotFound
+	}
+	err := db.client.Save(value).Error
+	if err != nil {
+		return errors.E(err, storage.ErrNotUpdated)
+	}
+	return nil
+}
+
+func (db *PgSql) Add(value interface{}) error {
```

Comment:
do you need to specify table name?  or how does it work?

Reply:
I did use Gorm, it's a Golang ORM. It automatically creates my databases based on the structs

https://gorm.io/

Reply:
I going start to implement and you going to see is easier to use